merge_streams_block.py

```diff
@@ -0,0 +1,54 @@
+from nio.common.block.attribute import Input
+from nio.common.block.base import Block
+from nio.common.discovery import Discoverable, DiscoverableType
+from nio.common.signal.base import Signal
+from nio.metadata.properties import VersionProperty, TimeDeltaProperty, \
+    BoolProperty
+from nio.modules.scheduler import Job
+
+
+@Input('input_2')
+@Input('input_1')
+@Discoverable(DiscoverableType.block)
+class MergeStreams(Block):
+
+    """ Take two input streams and combine signals together. """
+
+    expiration = TimeDeltaProperty(default={})
+    notify_once = BoolProperty(default=True)
+    version = VersionProperty('0.1.0')
+
+    def __init__(self):
+        super().__init__()
+        self._signals = { "input_1": {}, "input_2": {} }
+        self._expiration_jobs = { "input_1": None, "input_2": None }
+
+    def process_signals(self, signals, input_id='default'):
+        merged_signals = []
+        for signal in signals:
+            self._signals[input_id] = signal
+            if self._signals["input_1"] and self._signals["input_2"]:
+                merged_signals.append(self._merge_signals())
+        if self.expiration:
+            self._schedule_signal_expiration_job(input_id)
+        if merged_signals:
+            self.notify_signals(merged_signals)
+
+    def _merge_signals(self):
+        """ Merge signals 1 and 2 and clear from memory if only notify once """
+        merged_signal = self._signals["input_1"].to_dict()
```

Comment:
1. I think we have a bug with to_dict(hidden=True) because it looks like `{'C': 'c', '__doc__': ' A base signal class. No functionality to speak of\n\n    ', '__dict__': {'C': 'c'}, '__class__': <class 'nio.common.signal.base.Signal'>, '__weakref__': None, '__module__': 'nio.common.signal.base'}`

2. Does setting hidden=True do something weird with the way the dict references back to the Signal? Regardless of the dictionary including all those bad attrs, my tests are also failing from it seemingly updating the actual signal in `update`. (at least I think that's what is happening).

3. Continuing from my last comment, do you think it makes sense to do the to_dict and a deepcopy when saving the signal as opposed to when accessing it here?